addons/mail/static/src/messaging/entity/chat_window_manager/chat_window_manager.js

```diff
@@ -222,9 +185,95 @@ function ChatWindowManagerFactory({ Entity }) {
         }
 
         /**
+         * @param {mail.messaging.entity.ChatWindow} chatWindow
+         */
+        unregister(chatWindow) {
+            if (!this.allOrdered.includes(chatWindow)) {
+                return;
+            }
+            this.update({
+                _ordered: this._ordered.filter(
+                    _chatWindow => _chatWindow !== chatWindow.localId
+                ),
+                chatWindows: [['unlink', chatWindow]],
+            });
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * // FIXME: dependent on implementation that uses arbitrary order in relations!!
+         *
+         * @private
+         * @returns {mail.messaging.entity.ChatWindow}
+         */
+        _computeAllOrdered() {
+            return [['replace', this._ordered.map(_chatWindow => this.env.entities.ChatWindow.get(_chatWindow))]];
```

Comment:
A compute method that is returning a command instead of the result of the command? Feels strange